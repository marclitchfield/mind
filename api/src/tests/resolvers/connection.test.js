import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Connection tests', () => {
  applySetup();

  test('create connection', () => {
    return verifyMutations('Connection tests', 'create connection', {
      mutations: [
        ({space}) => gql`mutation { Person { person1:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({space}) => gql`mutation { Person { person2:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", datetime: "2010-07-17" title: "test person"}) { id } } }`,
        ({person1, person2, event}) => gql`mutation { Connection { connection:post(input:{person1:"${person1.id}" person2:"${person2.id}" eventId:"${event.id}" type:"union"}) { id } } }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            connections {
              participants { title }
              event { title }
            }
          }
        }
      }`
    });
  });

  test('remove connection', () => {
    return verifyMutations('Connection tests', 'remove connection', {
      mutations: [
        ({space}) => gql`mutation { Person { person1:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({space}) => gql`mutation { Person { person2:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", datetime: "2010-07-17" title: "test person"}) { id } } }`,
        ({person1, person2, event}) => gql`mutation { Connection { connection:post(input:{
          person1:"${person1.id}" person2:"${person2.id}" eventId:"${event.id}" type:"union"}) { id } } 
        }`,
        ({connection, person1, person2, event}) => gql`mutation { Connection { post(input:{
          remove: true, id:"${connection.id}" person1:"${person1.id}" person2:"${person2.id}" eventId:"${event.id}" type:"union"}) { id } } 
        }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          people {
            title
            connections {
              participants { title }
              event { title }
            }
          }
        }
      }`
    });
  });
});