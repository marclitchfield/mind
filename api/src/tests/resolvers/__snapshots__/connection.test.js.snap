// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection tests create connection 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "connections": Array [
          Object {
            "__typename": "Connection",
            "event": Object {
              "__typename": "Event",
              "title": "test person",
            },
            "participants": Array [
              Object {
                "__typename": "Person",
                "title": "test person",
              },
              Object {
                "__typename": "Person",
                "title": "test person",
              },
            ],
          },
        ],
        "title": "test person",
      },
      Object {
        "__typename": "Person",
        "connections": Array [
          Object {
            "__typename": "Connection",
            "event": Object {
              "__typename": "Event",
              "title": "test person",
            },
            "participants": Array [
              Object {
                "__typename": "Person",
                "title": "test person",
              },
              Object {
                "__typename": "Person",
                "title": "test person",
              },
            ],
          },
        ],
        "title": "test person",
      },
    ],
  },
}
`;

exports[`Connection tests remove connection 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "people": Array [
      Object {
        "__typename": "Person",
        "connections": Array [],
        "title": "test person",
      },
      Object {
        "__typename": "Person",
        "connections": Array [],
        "title": "test person",
      },
    ],
  },
}
`;
